src/hdmf/build/map.py

```diff
@@ -1127,9 +1130,18 @@ def __get_subspec_values(self, builder, spec, manager):
         elif isinstance(spec, DatasetSpec):
             if not isinstance(builder, DatasetBuilder):
                 raise ValueError("__get_subspec_values - must pass DatasetBuilder with DatasetSpec")
-            ret[spec] = builder.data
+            ret[spec] = self.__check_ref_resolver(builder.data)
         return ret
 
+    @staticmethod
+    def __check_ref_resolver(data):
```

Comment:
I'd rename this to ``__check_and_invert_ref_resolver`` it's a bit verbose, but I think will help with readability.